static/app/components/searchQueryBuilder/types.tsx

```diff
@@ -0,0 +1,28 @@
+export enum QueryBuilderFocusType {
+  FILTER_VALUE = 'filter_value',
+  FILTER_OP = 'filter_op',
+  FILTER_DELETE = 'filter_delete',
+}
+
+interface BaseTokenFocus {
```

Comment:
This is how we'll be determining which portion of the query is focused. We will store the selection range (character index of the query string). For filters we also store the part of the filter that is focused (key/op/value) and whether or not it is in edit mode.